From e33bd6bff84227828092cb349fa38d34f37af435 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Sun, 31 Jul 2016 11:13:38 +0300
Subject: [PATCH 0389/1345] eeprom: dts: a70x0, a80x0: add EEPROM device in DT
 for Armada-80/70x0

commit  9894286cfda645e3ea77161009eb312770bb16b2 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Add EEPROM "at24,24c64" device for Armada-80/70x0 DB
board to enable access to eeprom using device.
This will open the EEPROM as device and can access it using
Linux command under /sys/bus/i2c/devices/0-0057/eeprom

Change-Id: I17c65102f441d1e50725a956db91ee653cd67fe5
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/31540
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-7040-db.dtsi |   10 +++++++++
 arch/arm64/boot/dts/marvell/armada-8040-db.dtsi |   26 +++++++++++++++++++++++
 2 files changed, 36 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
index 284fcbf..72e21de 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
@@ -79,6 +79,16 @@
 			i2c1: i2c@701000 {
 				pinctrl-names = "default";
 				pinctrl-0 = <&i2c0_pins>;
+				eeprom@57 {
+					compatible = "at24,24c64";
+					reg = <0x57>;
+					pagesize = <32>;
+				};
+				eeprom@50 {
+					compatible = "at24,24c64";
+					reg = <0x50>;
+					pagesize = <32>;
+				};
 				expander0: pca9555@21 {
 					compatible = "nxp,pca9555";
 					pinctrl-names = "default";
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
index 5f4f2ab..2011292 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
@@ -64,8 +64,34 @@
 			i2c@511000 {
 				status = "okay";
 				clock-frequency = <100000>;
+				eeprom@57 {
+					compatible = "at24,24c64";
+					reg = <0x57>;
+					pagesize = <32>;
+				};
+				eeprom@50 {
+					compatible = "at24,24c64";
+					reg = <0x50>;
+					pagesize = <32>;
+				};
 			};
 		};
 	};
 
+	cpn-110-slave {
+		config-space {
+			i2c@701000 {
+				eeprom@57 {
+					compatible = "at24,24c64";
+					reg = <0x57>;
+					pagesize = <32>;
+				};
+				eeprom@50 {
+					compatible = "at24,24c64";
+					reg = <0x50>;
+					pagesize = <32>;
+				};
+			};
+		};
+	};
 };
-- 
1.7.9.5

